### PR TITLE
Fix compile error caused by moose.mk

### LINKED ISSUE #28698: `make` issues an error message related to `find`
### Bug Description

`make` issues the following error message at the very beginning:
```
find: illegal option -- m
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
```
While it doesn't prevent the compilation from proceeding, it ought to be fixed.

### Steps to Reproduce

```
cd framework
make
```

### Impact

This is an annoyance.

### [Optional] Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by hugary1995**
Do not use `find` when there is no non-unity directories.

close #28698

### CODE REVIEW

**Review by GiudGiud — DISMISSED** [review]

File: framework/moose.mk [hunk 1]
```diff
@@ -312,11 +312,12 @@ $(foreach srcsubdir,$(unity_srcsubdirs),$(eval $(call unity_file_rule,$(call uni
...
 app_unity_srcfiles := $(foreach srcsubdir,$(unity_srcsubdirs),$(call unity_unique_name,$(unity_src_dir),$(FRAMEWORK_DIR),$(srcsubdir)))
 
-#$(info $(app_unity_srcfiles))
-
 unity_srcfiles += $(app_unity_srcfiles)
 
-moose_srcfiles    := $(app_unity_srcfiles) $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+ifneq ($(non_unity_srcsubdirs),)
+	app_nonunity_srcfiles = $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+endif
```
Comment (GiudGiud) [inline framework/moose.mk:319]: what does find do when $(non_unity_srcsubdirs) is empty? seems it is finding some contrib here

Comment (GiudGiud) [reply framework/moose.mk:319]: i think we could have an "else" branch with no maxdepth used in the find and avoid the error

Comment (hugary1995) [reply framework/moose.mk:319]: When `non_unity_srcsubdirs` is empty, `find` emits the error message (which we are trying to fix here).

Comment (GiudGiud) [reply framework/moose.mk:319]: and it does not find anything? it just errors?

Comment (GiudGiud) [reply framework/moose.mk:319]: i guess so from local testing. umm so why are contribs not building

Comment (hugary1995) [reply framework/moose.mk:319]: I can't seem to reproduce the error locally.

Comment (hugary1995) [reply framework/moose.mk:319]: I mean some build boxes work, while others don't...

Comment (GiudGiud) [inline framework/moose.mk:319]:
```suggestion
else
  app_nonunity_srcfiles = $(shell find $(non_unity_srcsubdirs) -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
endif
```

Comment (GiudGiud) [reply framework/moose.mk:321]: trying something

Comment (hugary1995) [reply framework/moose.mk:321]: I assume you finished "trying something"... I'll look more into this now.

File: framework/moose.mk [hunk 2]
```diff
@@ -312,11 +312,14 @@ $(foreach srcsubdir,$(unity_srcsubdirs),$(eval $(call unity_file_rule,$(call uni
...
 
-#$(info $(app_unity_srcfiles))
-
 unity_srcfiles += $(app_unity_srcfiles)
 
-moose_srcfiles    := $(app_unity_srcfiles) $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+ifneq ($(non_unity_srcsubdirs),)
+	app_nonunity_srcfiles = $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+else
+  app_nonunity_srcfiles = $(shell find $(non_unity_srcsubdirs) -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
```
Comment (GiudGiud) [inline framework/moose.mk:320]:
```suggestion
  app_nonunity_srcfiles = $(shell find . -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
```

File: framework/moose.mk (hunk 1 above)
Comment (GiudGiud) [reply framework/moose.mk:321]: I think It works now. do you approve my 1 line change?

Comment (hugary1995) [reply framework/moose.mk:321]: sure -- if it works it works...

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**